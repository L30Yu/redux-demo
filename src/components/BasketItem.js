import React from "react";

export default function BasketItem({ order }) {
    return (
        <tr>
            <td>{order.name}</td>
            <td>{`@ $${order.cost} x ${order.count}`}</td>
            <td>{(order.cost * order.count).toFixed(2)}</td>
        </tr>
    );
}