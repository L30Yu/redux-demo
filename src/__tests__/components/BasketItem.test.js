import React from 'react';
import BasketItem from '../../components/BasketItem';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    let order = {
        id: 0,
        name: 'test',
        price: 2,
        count: 2
    }
    const tree = renderer
        .create(<BasketItem order={order} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
