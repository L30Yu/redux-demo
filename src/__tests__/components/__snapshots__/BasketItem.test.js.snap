// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    <div>
      test
       
      <br />
       
       
    </div>
  </td>
  <td>
    <div>
      @ $2 x 2
      <br />
       
    </div>
  </td>
  <td>
    <div>
      4.00
      <br />
       
    </div>
  </td>
</tr>
`;
